src/random.cpp

```diff
@@ -457,11 +406,21 @@ class RNGState {
         m_events_hasher.Write(events_hash, 32);
     }
 
+    /** Make the output of MixExtract (if allow_deterministic) deterministic, with specified seed. */
+    void MakeDeterministic(const uint256& seed) noexcept EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        LOCK(m_mutex);
+        m_deterministic_prng.emplace(MakeByteSpan(seed));
+    }
+
     /** Extract up to 32 bytes of entropy from the RNG state, mixing in new entropy from hasher.
      *
      * If this function has never been called with strong_seed = true, false is returned.
+     *
+     * If allow_deterministic is true, and MakeDeterministic has been called before, output
+     * from the deterministic PRNG is output instead.
```

Comment:
1. If I understand, you don't want to replace `MixExtract` entirely when the deterministic test RNG is used, because want `MixExtract` to still run and add/mix entropy in case someone decides to generate a keying material while the deterministic test RNG is on. This was what I was missing.

2. I like the name deterministic mode because it makes it more clear this is a RNG wide change.

3. Thought I had after waking up this morning was to change this to `bool fail_on_deterministic_mode`. Thus if someone had the deterministic test RNG activated and then tried to generate a key it would throw an error because `fail_on_deterministic_mode=true`. That said, I'm not sure this works with the current approach since if someone calls SeedStrengthen with the deterministic test RNG activated it would fail and it is reasonable to call SeedStrengthen in tests/fuzzing.